napari/layers/labels/labels.py

```diff
@@ -525,7 +542,54 @@ def paint(self, coord, new_label, refresh=True):
             slice_coord = tuple(slice_coord)
 
         # update the labels image
-        self.data[slice_coord] = new_label
+
+        if self._overwrite:
+            self.data[slice_coord] = new_label
+        else:
+            self.paint_without_overwrite(self.data, slice_coord, new_label, 0)
 
         if refresh is True:
             self.refresh()
+
+    def paint_without_overwrite(
+        self, array, slice_coord, new_label, dimension
+    ):
+        """
+        Paint the label without overwriting existing label, which is equivalent to only painting on background label.
+
+        Parameters
+        ----------
+        array : data array to paint on
+        slice_coord : coordinates to paint
+        new_label: new label to paint with
+        dimension: dimension of the current painting progress
```

Comment:
follow [numpy docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) for params:

```suggestion
        Parameters
        ----------
        array : np.ndarray
            Data array to paint on
        slice_coord : tuple
            Coordinates to paint
        new_label : int
            New label to paint with
        dimension : int
            Dimension of the current painting progress
```